File: app/src/main/java/com/github/kongpf8848/rxhttp/sample/MyApplication.java
package com.github.kongpf8848.rxhttp.sample;

import android.app.Application;
import android.util.Log;

import com.github.kongpf8848.rxhttp.RxHttpConfig;
import com.kongpf.commonhelper.ToastHelper;


public class MyApplication extends Application {

    @Override
    public void onCreate() {
        super.onCreate();
        Thread.setDefaultUncaughtExceptionHandler(new Thread.UncaughtExceptionHandler() {
            @Override
            public void uncaughtException(Thread t, Throwable e) {
                e.printStackTrace();
                Log.e("Exception", e.getMessage());
            }
        });

        ToastHelper.init(this);
        RxHttpConfig.getInstance()
                .maxRetries(3)
                .retryDelayMillis(200)
                .debugMode(BuildConfig.DEBUG);


    }
}

File: app/src/main/java/com/github/kongpf8848/rxhttp/sample/MyApplication.kt
package com.github.kongpf8848.rxhttp.sample

import android.app.Application
import com.github.kongpf8848.rxhttp.FixHttpLoggingInterceptor
import com.github.kongpf8848.rxhttp.RxHttpConfig
import com.github.kongpf8848.rxhttp.sample.http.interceptor.MockInterceptor
import com.github.kongpf8848.rxhttp.sample.utils.LogUtils
import com.kongpf.commonhelper.ToastHelper


class MyApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        Thread.setDefaultUncaughtExceptionHandler { t, e ->
            e.printStackTrace()
            LogUtils.e("Crash", e.message)
        }
        LogUtils.init(this, BuildConfig.DEBUG)
        ToastHelper.init(this)

        /**
         * 对RxHttp进行配置，可选
         * 失败重试3次，每次间隔200毫秒
         * 添加日志打印拦截器HttpLoggingInterceptor
         * 添加模拟网络请求拦截器MockInterceptor
         */
        RxHttpConfig.getInstance()
                .maxRetries(3)
                .retryDelayMillis(200)
                .getBuilder().apply {
                    if (BuildConfig.DEBUG) {
                        addInterceptor(FixHttpLoggingInterceptor().apply {
                            level = FixHttpLoggingInterceptor.Level.BODY
                        })
                        addInterceptor(MockInterceptor())
                    }
                }

    }
}